Syncing latest changes from upstream devel for ceph-csi

df-build-team opened the pull request:
PR containing the latest commits from upstream devel branch

Commit:
rbd: support nbd on euler or arm

Signed-off-by: muxuelan <muxuelan@cmss.chinamobile.com>

nixpanic committed:
vendor: include k8s.io/api/rbac/v1

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
api: add CSIProvisionerRBAC functions for the NFS-provisioner

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
build: add NFS provisioner RBAC to generated artifacts

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
rebase: update vendored Ceph-CSI API

Signed-off-by: Niels de Vos <ndevos@ibm.com>

iPraveenParihar committed:
kms: Implement Azure key vault as KMS provider

This commit adds the Azure Key Vault as a supported
KMS provider.

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar committed:
doc: added docs for Azure KMS

Signed-off-by: Praveen M <m.praveen@ibm.com>

iPraveenParihar committed:
rebase: Azure key vault module dependency update

This commit adds the Azure SDK for Azure key vault KMS
integration to the Ceph CSI driver.

Signed-off-by: Praveen M <m.praveen@ibm.com>

dragoangel committed:
doc: Update capabilities readme to solve to open permissions

Signed-off-by: Dmytro Alieksieiev <1865999+dragoangel@users.noreply.github.com>

dependabot[bot] committed:
rebase: bump google.golang.org/grpc from 1.62.0 to 1.62.1

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.62.0 to 1.62.1.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.62.0...v1.62.1)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

nixpanic committed:
cleanup: don't return an internal type from VolumeGroupJournal.Connect()

The VolumeGroupJournal interface does not need to return anything except
for a potential error. Any instance that implements the
VolumeGroupJournal interface can be used to call all functions.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

dependabot[bot] committed:
rebase: bump google.golang.org/protobuf in /actions/retest

Bumps google.golang.org/protobuf from 1.31.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

nixpanic commented:
/assign iPraveenParihar
/approve

Please check if this is all that you need for the Azure KMS feature, and `/lgtm` it when it looks good.

nixpanic committed:
cleanup: correct typo in `NewCSIVolumeroupJournal()` function

The name of the function should be `NewCSIVolumeGroupJournal()`.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
cleanup: do not pass EncodingVersion to `GenerateVolID()`

The only encoding version that exists is `1`. There is no need to have
multiple constants for that version across different packages. Because
there is only one version, `GenerateVolID()` does not really require it,
and it can use a default version.

If there is a need in the future to support an other encoding version,
this can be revisited with a cleaner solution.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
e2e: replace quoted namespace in templates too

Some templates are now generated with the API, and these include
namespaces as "quotes" values. Namespace replacing in the templates need
to replace both the unquoted and quoted strings.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
deploy: include ServiceAccount in the NFS provisioner RBAC artifact

It seems that the ServiceAccount was not created anymore, this causes
problems with provisioning volumes.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Madhu-1 commented:
/lgtm
